device/cuda/CMakeLists.txt

```diff
@@ -44,10 +44,18 @@ traccc_add_library( traccc_cuda cuda TYPE SHARED
   # Fitting
   "include/traccc/cuda/fitting/fitting_algorithm.hpp"
   "src/fitting/fitting_algorithm.cu")
-  
+if(TRACCC_SETUP_PROFILER)
 target_compile_options( traccc_cuda
-  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr> )
+  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr -Xcompiler -finstrument-functions -fPIC > )
 target_link_libraries( traccc_cuda
-  PUBLIC traccc::core detray::core detray::utils vecmem::core covfie::core
-  PRIVATE CUDA::cudart traccc::device_common vecmem::cuda )
+        PUBLIC traccc::core detray::core detray::utils vecmem::core covfie::core
+        PRIVATE CUDA::cudart traccc::device_common vecmem::cuda -lnvToolsExt)
```

Comment:
The CMake documentation is unclear about whether your current implementation would work but it would be more conformant to the specification (although syntactically less clear) to write ` nvToolsExt` instead of `-lnvToolsExt`.

Reply:
I have implemented your request